service/dap/server.go

```diff
@@ -97,26 +97,26 @@ var defaultArgs = launchAttachArgs{
 
 // DefaultLoadConfig controls how variables are loaded from the target's memory,
 // borrowing the default value from the original vscode-go debug adapter and rpc server.
-// For evaluate requests made in the context of 'repl' or 'variables', they are triggered
-// by explicit human action so we use InteractiveEvaluateLoadConfig to apply larger limits.
 // TODO(polina): Support setting config via launch/attach args or only rely on on-demand loading?
-var (
-	DefaultLoadConfig = proc.LoadConfig{
-		FollowPointers:     true,
-		MaxVariableRecurse: 1,
-		MaxStringLen:       64,
-		MaxArrayValues:     64,
-		MaxStructFields:    -1,
-	}
+var DefaultLoadConfig = proc.LoadConfig{
+	FollowPointers:     true,
+	MaxVariableRecurse: 1,
+	MaxStringLen:       64,
+	MaxArrayValues:     64,
+	MaxStructFields:    -1,
+}
 
-	InteractiveEvaluateLoadConfig = proc.LoadConfig{
-		FollowPointers:     true,
-		MaxVariableRecurse: 64,
-		MaxStringLen:       4096,
-		MaxArrayValues:     4096,
-		MaxStructFields:    -1,
-	}
-)
+// For evaluate requests made in the context of 'repl' or 'variables', they are triggered
+// by explicit human action and the results can be displayed in a larger space (e.g. DEBUG CONSOLE)
+// or copied in memory, so we use InteractiveEvaluateLoadConfig to apply larger limits
+// if the result is represented as a string. We won't apply this heuristic to `call` command.
+var InteractiveEvaluateLoadConfig = proc.LoadConfig{
+	FollowPointers:     true,
+	MaxVariableRecurse: 1,
+	MaxStringLen:       1024,
+	MaxArrayValues:     1024,
```

Comment:
The commit name says that you are applying larger limit to string only, but you are also applying it to arrays. Is this intentional? Also, there is no need to match them. Extra characters should be much cheaper than extra objects.

Reply:
Outdated code.